### ISSUE TITLE
Feat: Undo

### ISSUE THREAD

**Issue by knudsvik** [equinor/flyt#295]
Acceptance Criteria:
Undo 3 steps back for now

Track state-changing user actions.
Save the previous state before each action.
Allow the client to request an “undo,” which rolls back to an earlier state.

Testing Required:
Yes, acceptance criteria is to be fulfilled.

Definition of Done:

**User input 1 (by LSE, GLB, HE):**
Sensitivt - klarer både å slette og flytte bokser uten advarsel => få en undo knapp?

**User input 2 (by LH, feedback ID 208 ):**
Can be nice to be able to regrett. I.e regrett adding a new card. Or regrett writing/deleting something

**User input 3 (by FL, feedback ID 230 ):**
As a user of Flyt I want to have the option to undo (ctrl+z??) when I (frequently) do things I would like to revert.

**User input 4 (by SEG, feedback ID 237 ):**
As a user I want to undo the last step so that I can correct mistakes. Deleted something and cannot retrieve.
No button or option to retrieve or undo. (Sent in via Service Now, added manually to Feedback list)

**User story**
As a user I want to undo the last step so that I can correct mistakes

**Current design proposal**
<img width="794" alt="Skjermbilde 2021-11-11 kl  11 44 52" src="https://user-images.githubusercontent.com/74361011/141284809-63974165-8c95-49a0-9c62-0c12e5ecc501.png">

**Comment by knudsvik** [comment]
Dependent on #154 according to @SjoenH

**Comment by HavardNot** [comment]
Might we use React Flow Pro for this? I see they have an example implementation for undo/redo functionality:
https://reactflow.dev/examples/interaction/undo-redo

**Comment by KaterynaBasik** [comment]
From the echnical point of view, the implementation of saving process versions and states for all users is possible, but requires significant resources. Parked for now

**Comment by jsundarn** [comment]
Design is updated [here](https://www.figma.com/design/nqTjMBJ1NDeAN8UaZ4gKR8/Flyt-Production?node-id=273-9908&t=TVkEnJlvxYIN0YsX-1).

@askbulle @AnjaliPandit20 @KarianneL

**Comment by AnjaliPandit20** [comment]
@jsundarn @srvEq : Undo 3 steps back for now

**Comment by MBanova** [comment]
Acceptance Criteria: 
Undo 3 steps back for now

Track state-changing user actions.
Save the previous state before each action.
Allow the client to request an “undo,” which rolls back to an earlier state.

Testing Required:
Yes, acceptance criteria is to be fulfilled.

Definition of Done:

**Comment by MBanova** [comment]
waiting a response from the customer (https://github.com/equinor/flyt/issues/954)

**Comment by MBanova** [comment]
Feature development is finished, and bugs found after complex testing should be fixed.

**Comment by MBanova** [comment]
deploy on dev for testing
